app/src/pages/Play/operations.graphql

```diff
@@ -23,6 +26,7 @@ mutation EndCurrentTurnAndStartNextTurn(
   $gameId: uuid!
   $nextTurnplayerId: uuid!
   $nextTurnSecondsPerTurnOverride: Int
+  $roundId: uuid
```

Comment:
I originally put this as `uuid!`, but in the current implementation I don't think this is right.

It looks like we call `EndCurrentTurnAndStartNextTurn` at the end of every turn (including the last one), which means we always have one zombie turn at the end of every game. I don't think we should set a `round_id` on that turn (since it doesn't belong to a round), which means this (and the db column) needs to be nullable.

Perhaps in a future refactor we could not create a new turn if we detect the end game condition. It looks like graphql mutations don't support conditional logic, so I'm not sure the best pattern for handling that.

Reply:
Ah nice catch. makes sense. I think the best pattern would be to use a different mutation. say EndCurrentTurn (that doesn't do the StartNextTurn part). Alternatively, split out StartNextTurn as a separate mutation. We can do this as an additional body of work tho. Makes typing a bit harder, but just requires some pre processing (filter out all turns that don't have a round id, they are irrelevant anyways)

Reply:
Gotcha, yeah I'm happy to do as a follow-up. I don't have a great sense about the most idiomatic way to do it, so we can chat more in the future.

Reply:
More I think about it - I think idiomatic way would be creating a custom action. Lots going on here, maybe useful to have one endpoint - that can deal w/the conditionals appropriately.